packages/core-accountsjs/core-accountsjs.js

```diff
@@ -0,0 +1,205 @@
+import { AccountsServer, ServerHooks, AccountsJsError } from '@accounts/server';
+import { AccountsPassword } from '@accounts/password';
+import MongoDBInterface from '@accounts/mongo';
+import { MongoInternals } from 'meteor/mongo';
+import { DatabaseManager } from '@accounts/database-manager';
+import { Random } from 'meteor/random';
+import crypto, { randomBytes } from 'crypto';
+import defer from 'lodash.defer';
+
+// import { destroyToken, hashLoginToken } from './util';
+
+const CreateUserErrors = {
+  /**
+   * Will throw if no username or email is provided.
+   */
+  UsernameOrEmailRequired: 'UsernameOrEmailRequired',
+  /**
+   * Username validation via option `validateUsername` failed.
+   */
+  InvalidUsername: 'InvalidUsername',
+  /**
+   * Email validation via option `validateEmail` failed.
+   */
+  InvalidEmail: 'InvalidEmail',
+  /**
+   * Password validation via option `validatePassword` failed.
+   */
+  InvalidPassword: 'InvalidPassword',
+  /**
+   * Email already exist in the database.
+   */
+  EmailAlreadyExists: 'EmailAlreadyExists',
+  /**
+   * Username already exist in the database.
+   */
+  UsernameAlreadyExists: 'UsernameAlreadyExists',
+};
+
+const METEOR_ID_LENGTH = 17;
+
+export const idProvider = () =>
+  randomBytes(30)
+    .toString('base64')
+    .replace(/[\W_]+/g, '')
+    .substr(0, METEOR_ID_LENGTH);
+
+const dateProvider = (date) => date || new Date();
+
+const mongoStorage = new MongoDBInterface(
+  MongoInternals.defaultRemoteCollectionDriver().mongo.db,
+  {
+    convertUserIdToMongoObjectId: false,
+    convertSessionIdToMongoObjectId: false,
+    idProvider,
+    dateProvider,
+  },
+);
+
+const dbManager = new DatabaseManager({
+  sessionStorage: mongoStorage,
+  userStorage: mongoStorage,
+});
+
+const accountsServerOptions = {
+  siteUrl: Meteor.absoluteUrl(),
+  // sendVerificationEmailAfterSignup: false, // It's set to false by default
+};
+
+class UnchainedAccountsPassword extends AccountsPassword {
+  async createUser(user, options) {
```

Comment:
Can you explain why you overwrite createUser?

Reply:
Because the original implementation doesn't allow use to pass the options argument which we later need in our hooks.
https://github.com/accounts-js/accounts/blob/master/packages/password/src/accounts-password.ts#L593

Reply:
Hm, I see...

As it's class based, maybe you could you assign the context to the AccountServer in a way that it's available on "this", so you could go like this.unchainedContext in these methods. Maybe that could work so we could rely on the accountsjs code?

Reply:
Another thought: skipping e-mail verifications when a guest user is created could propably be skipped at a later time? maybe in server.prepareEmail or server.sendMail (we will have to touch these anyway later in the process)

Reply:
Good point on passing `this.unchainedContext`. I've had a closer look there's [sendVerificationEmail](https://github.com/accounts-js/accounts/blob/master/packages/password/src/accounts-password.ts#L650) option but the only downside to it is that we can't fine tune it per request. Yet, if it's ok with you we can actually work with it.